test/method.jl

```diff
@@ -0,0 +1,181 @@
+macro test_signature(function_def_expr, method=nothing)
+    _target = splitdef(function_def_expr)
+    return quote
+        fun = $(esc(function_def_expr))
+        m = if ($method === nothing)
+            only_method(fun)
+        else
+            $(esc(method))
+        end
+        sig = signature(m)
+        test_matches(sig, $(_target))
+    end
+end
+
+function test_matches(candidate::AbstractDict, target::Dict)
+    # we want to use literals in the tests so that @test gives useful output on failure
+    haskey(target, :name) && @test target[:name] == get(candidate, :name, nothing)
+    haskey(target, :params) && @test target[:params] == get(candidate, :params, nothing)
+    haskey(target, :args) && @test target[:args] == get(candidate, :args, nothing)
+    haskey(target, :whereparams) &&
+        @test target[:whereparams] == get(candidate, :whereparams, nothing)
```

Comment:
Should we add tests for the other fields that `splitdef` returns and just use `@test_broken`?

Reply:
I don't think it is required. Do you think it is?
Right now we are matching what we have documented it does I would.
Should we  save test broken for actual bugs?

Reply:
We should do `@test !haskey(candidate, :rtype)` to ensure these aren't set which is what we expect. If someone attempts to get these working then the test will fail and they will need to update these tests.

Reply:
i became convinced and added `@test_broken`